packages/language-support/src/autocompletion/completionCoreCompletions.ts

```diff
@@ -331,14 +332,33 @@ const parameterCompletions = (
     )
     .map(([paramName]) => {
       const backtickedName = backtickIfNeeded(paramName);
+      let maybeInsertText = backtickedName
+        ? { insertText: `$${backtickedName}` }
+        : {};
+      if (previousToken) {
+        const suffix = maybeInsertText.insertText ?? `$${paramName}`;
+        maybeInsertText = {
+          // We need to complete parameters correctly in VSCode,
+          // otherwise when we have 'RETURN $' and we get offered $param
+          // we would complete RETURN $$param, which is not what we want
+          insertText: computeSuffix(previousToken.text, suffix),
```

Comment:
Probably this needs tests in the language support side and the react codemirror side